music-quiz/src/presenters/userProfilePresenter.js

```diff
@@ -41,6 +42,32 @@ export default function UserProfilePresenter(props) {
     return whenTakenDown;
   }
 
+  function averageRating(playlistHistory) {
+    let ratings = Object.values(playlistHistory).filter(
+      (history) => history.score
+    );
+    if (ratings.length === 0) {
+      return <p style={{ margin: "0px" }}>No Rating</p>;
+    }
+    let sumOfRatings = 0;
+    ratings.map((score, index) => {
+      sumOfRatings += score.rating;
+    });
+    let averageRating = Math.round(sumOfRatings / ratings.length);
+    return (
+      <div style={{ display: "flex" }}>
+        <p style={{ margin: "3px 3px" }}>Rating:</p>
+        <ReactStars
+          count={5}
+          value={averageRating}
+          size={20}
+          activeColor="#ffd700"
+          edit={false}
+        />
+      </div>
+    );
+  }
```

Comment:
If a playlist does not have any scores/ratings (the `/playerHistory` directory does not exist in firebase) then this code crashes.

Reply:
We fixed this by adding a text saying "No ratings" and "This playlist has not been played yet" so now it works when no one has played it yet.